.tools/validate.go

```diff
@@ -0,0 +1,269 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"regexp"
+	"strings"
+)
+
+// DefaultRules are the standard validation to perform on git commits
+var DefaultRules = []ValidateRule{
+	func(c CommitEntry) (vr ValidateResult, err error) {
+		vr.CommitEntry = c
+		hasValid := false
+		for _, line := range strings.Split(c.Body, "\n") {
+			if validDCO.MatchString(line) {
+				hasValid = true
+			}
+		}
+		if !hasValid {
+			vr.Pass = false
+			vr.Msg = "does not have a valid DCO"
+		} else {
+			vr.Pass = true
+			vr.Msg = "has a valid DCO"
+		}
+
+		return vr, nil
+	},
+	// TODO add something for the cleanliness of the c.Subject
+}
+
+var (
+	flVerbose     = flag.Bool("v", false, "verbose")
+	flCommitRange = flag.String("range", "", "use this commit range instead")
+
+	validDCO = regexp.MustCompile(`^Signed-off-by: ([^<]+) <([^<>@]+@[^<>]+)>$`)
+)
+
+func main() {
+	flag.Parse()
+
+	var commitrange string
+	if *flCommitRange != "" {
+		commitrange = *flCommitRange
+	} else {
+		var err error
+		commitrange, err = GitFetchHeadCommit()
+		if err != nil {
+			log.Fatal(err)
+		}
+	}
+
+	c, err := GitCommits(commitrange)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	results := ValidateResults{}
+	for _, commit := range c {
+		fmt.Printf(" * %s %s ... ", commit.AbbreviatedCommit, commit.Subject)
+		if vr, err := ValidateCommit(commit, DefaultRules); err != nil {
+			log.Fatalf("%s: %s", commit.Commit, err)
```

Comment:
Ah, nevermind.  If we [drop the error-code][1], this block will go away entirely.

[1]: https://github.com/opencontainers/specs/pull/167/files#r39100899